scipy/sparse/meson.build

```diff
@@ -5,7 +5,8 @@ _csparsetools_pyx = custom_target('_csparsetools_pyx',
 )
 
 py3.extension_module('_csparsetools',
-  _csparsetools_pyx,
+  cython_gen.process(_csparsetools_pyx),
+  c_args: ['-Wno-cpp'],
```

Comment:
Add the comment here as well?

# To avoid deprecated Numpy API warning

Reply:
Yes, I agree - it would be better to signal that more consistently.  What do you think about 

```
c_old_numpy_api = ['-Wno-cpp']
```

and 

```
c_args : c_old_numpy_api
```

througout?

Reply:
There is already a standard comment:
```
['-Wno-cpp'], # TODO: replace with numpy_nodepr_api
```

Reply:
In one place I expanded it to:
```
# Replace with numpy_nodepr_api after Cython 3.0 is out
```
which is the most informative thing.

Reply:
My proposal was to put that comment next to a suitably named variable, and then let people go to the variable to work out what's going on, as in:

```
# Suppress warning for deprecated Numpy API.
# Replace with numpy_nodepr_api after Cython 3.0 is out
c_no_warn_dep_np_api = ['-Wno-cpp']
```

or similar.  But happy to copy paste same comment for each `-Wno-cpp` if you prefer.

Reply:
Either way is fine, I don't have a strong preference. If you do want to use `c_no_warn_dep_np_api`, please change it everywhere though.